pkg/controller/master/virtualmachine/vmi_controller.go

```diff
@@ -91,3 +105,41 @@ func (h *VMIController) UnsetOwnerOfPVCs(_ string, vmi *kubevirtv1.VirtualMachin
 
 	return vmi, nil
 }
+
+// ReconcileFromHostLabels handles the propagation of metadata from node labels to VirtualMachineInstance annotations.
+func (h *VMIController) ReconcileFromHostLabels(_ string, vmi *kubevirtv1.VirtualMachineInstance) (*kubevirtv1.VirtualMachineInstance, error) {
+	if vmi == nil || vmi.DeletionTimestamp != nil {
+		return vmi, nil
+	}
+
+	if creator := vmi.Labels[builder.LabelKeyVirtualMachineCreator]; creator != VirtualMachineCreatorNodeDriver {
+		return vmi, nil
+	}
+
+	node, err := h.nodeCache.Get(vmi.Status.NodeName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return vmi, nil
+		}
+		return vmi, fmt.Errorf("failed to get node %s for VirtualMachineInstance %s: %v", vmi.Status.NodeName, vmi.Name, err)
```

Comment:
Better to add a log when error returning here.

Reply:
updated

Reply:
Here is a backport PR, I can't change it directly, so create a new PR after the merge to change it.

Reply:
Just to emphasize here that we don't change the code in the backport PR, if needed feel free to file a new PR to make an enhancement, thanks.